### DISCUSSION TITLE
Adding out-of-stock items to cart and checking out does not throw error on server. How to handle inventory levels?

### DISCUSSION THREAD

**Discussion by tannergaucher** [Shopify/storefront-api-feedback#49] [Help]
I noticed that I can add a product to my cart and checkout where the `selectedVariant` object has values of `currentlyNotInStock: true` and `quantityAvailable: 0`. I expected the `cartLineAdd` mutation to not send a 200 response on an out-of-stock product. And the same for checking out.

Currently I'm disabling buttons on the client based on the `currentlyNotInStock` boolean, but the situation requires more than a client-only solution, as in a high-volume scenario inventory levels, price, etc. may likely have changed since a user loaded the page.

Does Shopify not handle product inventory levels server-side? 

What's the best way to ensure that we are not allowing customers to purchase items that are not in stock?

**Comment by patjakubik** [comment]
Duplicate of #40
